raft/src/main/java/org/apache/kafka/raft/ResignedState.java

```diff
@@ -125,6 +131,12 @@ public long remainingElectionTimeMs(long currentTimeMs) {
         return preferredSuccessors;
     }
 
+    @Override
+    public boolean grantVote(int candidateId, Supplier<Boolean> logComparator) {
+        log.debug("Rejecting vote request since we have resigned as candidate/leader in this epoch");
+        return false;
```

Comment:
I'm sure there must have been a reason for this in the original implementation, but it still surprised me to see it. I think it makes sense for the resigned leader to help another candidate get elected. I'll file a separate JIRA about this.

Reply:
Filed this one: https://issues.apache.org/jira/browse/KAFKA-12607.